apollo-router/tests/integration/directives/policy.rs

```diff
@@ -0,0 +1,604 @@
+    let coprocessor_address = mock_coprocessor.uri();
+
+    Mock::given(method("POST"))
+        .and(path("/"))
+        .respond_with(ResponseTemplate::new(200).set_body_json(serde_json::json!({
+            "version": 1,
+            "stage": "SupergraphRequest",
+            "control": "continue",
+            "context": {
+                "entries": {
+                    "apollo::authorization::required_policies": {
+                        // NOTE: see the note above, but this shouldn't govern how the test
+                        // behaves; it's the context object that does the dirt for the TestHarness.
+                        // Change this value if you use it with the IntegrationTest builder
+                        "admin": true
+                    }
+                }
+            }
+        })))
+        .mount(&mock_coprocessor)
+        .await;
+
+    let mut subgraphs = MockedSubgraphs::default();
+    subgraphs.insert(
+        "subgraph_a",
+        MockSubgraph::builder()
+            .with_json(
+                serde_json::json!({"query": "{private{id}}"}),
+                serde_json::json!({"data": {"private": {"id": "123"}}}),
+            )
+            .with_json(
+                serde_json::json!({"query": "{public{id}}"}),
+                serde_json::json!({"data": {"public": {"id": "456"}}}),
+            )
+            .with_json(
+                serde_json::json!({"query": "{opensecret{id}}"}),
+                serde_json::json!({"data": {"opensecret": {"id": "789"}}}),
+            )
+            .with_json(
+                serde_json::json!({"query": "{secure{id}}"}),
+                serde_json::json!({"data": {"secure": {"id": "000"}}}),
+            )
+            .build(),
+    );
+
+    let supergraph_harness = TestHarness::builder()
+        .configuration_json(serde_json::json!({
+            "coprocessor": {
+                "url": coprocessor_address,
+                "supergraph": {
+                    "request": {
+                        "context": "all"
+                    }
+                }
+            },
+            "include_subgraph_errors": {
+                "all": true
+            }
+        }))
+        .unwrap()
+        .schema(include_str!(
+            "../../fixtures/directives/policy/policy_schema_with_interfaces.graphql"
+        ))
+        .extra_plugin(subgraphs)
+        .build_supergraph()
+        .await
+        .unwrap();
+
+    let context = Context::new();
+    context
+        .insert(
+            "apollo::authorization::required_policies",
+            json! {[ "admin" ]},
+        )
+        .unwrap();
+
+    // WHEN
+    //   * we make a request with the interface itself off of Query
+    let request = supergraph::Request::fake_builder()
+        .query(r#"{ secure { id } }"#)
+        .context(context)
+        .method(Method::POST)
+        .build()
+        .unwrap();
+
+    // THEN
+    //   * we don't get any data!
```

Comment:
we get errors though, right? if so, this test should check for that error.

Reply:
turns out we do; I didn't realize that, check added